<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1024</width>
    <height>555</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>1024</width>
    <height>555</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1024</width>
    <height>555</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="GLWidget" name="glwidget" native="true">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>0</y>
      <width>704</width>
      <height>528</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">GLWidget {
	border: 1px solid black;
}</string>
    </property>
   </widget>
   <widget class="QFrame" name="frame_rotation_sliders">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>200</y>
      <width>320</width>
      <height>85</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QFrame {
	background-color: gray;
}</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::NoFrame</enum>
    </property>
    <widget class="QLabel" name="label_rotate_y">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>30</y>
       <width>12</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Y</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_rotate_z">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>50</y>
       <width>12</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Z</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_rotate_x">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>10</y>
       <width>12</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>X</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="rotating_y_value">
     <property name="geometry">
      <rect>
       <x>275</x>
       <y>30</y>
       <width>35</width>
       <height>17</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="rotating_z_value">
     <property name="geometry">
      <rect>
       <x>275</x>
       <y>50</y>
       <width>35</width>
       <height>17</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="rotating_x_value">
     <property name="geometry">
      <rect>
       <x>275</x>
       <y>10</y>
       <width>35</width>
       <height>17</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_rotate">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>40</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>rotate:</string>
     </property>
    </widget>
    <widget class="QScrollBar" name="slider_x_rotation">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>10</y>
       <width>200</width>
       <height>16</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QScrollBar" name="slider_y_rotation">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>30</y>
       <width>200</width>
       <height>16</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QScrollBar" name="slider_z_rotation">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>50</y>
       <width>200</width>
       <height>16</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="frame_scale">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>70</y>
      <width>320</width>
      <height>50</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="styleSheet">
     <string notr="true">QFrame {
	background-color: gray;
}</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::NoFrame</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLineEdit" name="scale_text">
     <property name="geometry">
      <rect>
       <x>275</x>
       <y>7</y>
       <width>35</width>
       <height>17</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_scale">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>6</y>
       <width>41</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>scale:</string>
     </property>
    </widget>
    <widget class="QScrollBar" name="scale_slider">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>7</y>
       <width>200</width>
       <height>16</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="frame_settings">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>280</y>
      <width>320</width>
      <height>100</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QFrame {
	background-color: gray;
}</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::NoFrame</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLabel" name="label_bar_thickness">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>30</y>
       <width>40</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>thicc:</string>
     </property>
    </widget>
    <widget class="QScrollBar" name="slider_edge_thickness">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>32</y>
       <width>105</width>
       <height>20</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QComboBox" name="comboBox_typeLine">
     <property name="geometry">
      <rect>
       <x>5</x>
       <y>30</y>
       <width>150</width>
       <height>32</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>edges: solid</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>edges: dashed</string>
      </property>
     </item>
    </widget>
    <widget class="QComboBox" name="comboBox_typeVertex">
     <property name="geometry">
      <rect>
       <x>5</x>
       <y>55</y>
       <width>150</width>
       <height>32</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>verticles: none</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>verticles: round</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>verticles: square</string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="label_bar_size">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>55</y>
       <width>40</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>size:</string>
     </property>
    </widget>
    <widget class="QScrollBar" name="slider_vertice_size">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>57</y>
       <width>105</width>
       <height>20</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QComboBox" name="comboBox_projection">
     <property name="geometry">
      <rect>
       <x>5</x>
       <y>5</y>
       <width>310</width>
       <height>32</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>projection: central / perspective</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>projection: parallel / orthogonal</string>
      </property>
     </item>
    </widget>
   </widget>
   <widget class="QFrame" name="frame_reset">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>460</y>
      <width>320</width>
      <height>40</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QFrame {
	background-color: gray;
}</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::NoFrame</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QPushButton" name="reset_values_button">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>0</y>
       <width>300</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>reset values</string>
     </property>
     <property name="autoDefault">
      <bool>false</bool>
     </property>
     <property name="default">
      <bool>false</bool>
     </property>
     <property name="flat">
      <bool>false</bool>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="frame_colors">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>380</y>
      <width>320</width>
      <height>85</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QFrame {
	background-color: gray;
}</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::NoFrame</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QPushButton" name="color_verticies_button">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>0</y>
       <width>100</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>verticles</string>
     </property>
    </widget>
    <widget class="QPushButton" name="color_edges_button">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>0</y>
       <width>100</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>edges</string>
     </property>
     <property name="shortcut">
      <string>Backspace</string>
     </property>
    </widget>
    <widget class="QPushButton" name="background_color_button">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>0</y>
       <width>100</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>background</string>
     </property>
    </widget>
    <widget class="QGraphicsView" name="graphics_verticies_color">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>27</y>
       <width>100</width>
       <height>25</height>
      </rect>
     </property>
    </widget>
    <widget class="QGraphicsView" name="graphics_color_edges">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>27</y>
       <width>100</width>
       <height>25</height>
      </rect>
     </property>
    </widget>
    <widget class="QGraphicsView" name="graphics_background_color">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>27</y>
       <width>100</width>
       <height>25</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="frame_count">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>500</y>
      <width>320</width>
      <height>30</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="styleSheet">
     <string notr="true">QFrame {
	background-color: gray;
}</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::NoFrame</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLabel" name="label_count_of_verticies">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>6</y>
       <width>61</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>verticies:</string>
     </property>
    </widget>
    <widget class="QLabel" name="number_of_vertices">
     <property name="geometry">
      <rect>
       <x>75</x>
       <y>6</y>
       <width>75</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_count_of_facets">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>6</y>
       <width>48</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>edges:</string>
     </property>
    </widget>
    <widget class="QLabel" name="number_of_facets">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>6</y>
       <width>75</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="frame_buttons">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>320</width>
      <height>75</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="styleSheet">
     <string notr="true">QFrame {
	background-color: gray;
}</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::NoFrame</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QPushButton" name="open_button">
     <property name="geometry">
      <rect>
       <x>115</x>
       <y>5</y>
       <width>195</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>open</string>
     </property>
     <property name="default">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="bmp_button">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>30</y>
       <width>90</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>bmp</string>
     </property>
    </widget>
    <widget class="QPushButton" name="gif_button">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>90</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>gif</string>
     </property>
    </widget>
    <widget class="QPushButton" name="jpeg_button">
     <property name="geometry">
      <rect>
       <x>115</x>
       <y>30</y>
       <width>90</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>jpeg</string>
     </property>
    </widget>
    <widget class="QPushButton" name="rec_button">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>5</y>
       <width>90</width>
       <height>28</height>
      </rect>
     </property>
     <property name="text">
      <string>recording</string>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="frame_moving_sliders">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>120</y>
      <width>320</width>
      <height>85</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QFrame {
	background-color: gray;
}</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::NoFrame</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLabel" name="label_move_x">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>10</y>
       <width>12</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>X</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_move_y">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>30</y>
       <width>12</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Y</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_move_z">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>50</y>
       <width>12</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Z</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="moving_x_value">
     <property name="geometry">
      <rect>
       <x>275</x>
       <y>10</y>
       <width>35</width>
       <height>17</height>
      </rect>
     </property>
     <property name="layoutDirection">
      <enum>Qt::LeftToRight</enum>
     </property>
    </widget>
    <widget class="QLineEdit" name="moving_y_value">
     <property name="geometry">
      <rect>
       <x>275</x>
       <y>30</y>
       <width>35</width>
       <height>17</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="moving_z_value">
     <property name="geometry">
      <rect>
       <x>275</x>
       <y>50</y>
       <width>35</width>
       <height>17</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_move">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>35</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>move:</string>
     </property>
    </widget>
    <widget class="QScrollBar" name="slider_x_moving">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>10</y>
       <width>200</width>
       <height>16</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QScrollBar" name="slider_y_moving">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>30</y>
       <width>200</width>
       <height>16</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QScrollBar" name="slider_z_moving">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>50</y>
       <width>200</width>
       <height>16</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </widget>
   <zorder>frame_reset</zorder>
   <zorder>frame_scale</zorder>
   <zorder>glwidget</zorder>
   <zorder>frame_settings</zorder>
   <zorder>frame_rotation_sliders</zorder>
   <zorder>frame_colors</zorder>
   <zorder>frame_count</zorder>
   <zorder>frame_buttons</zorder>
   <zorder>frame_moving_sliders</zorder>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="action">
   <property name="text">
    <string>Open File...</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>GLWidget</class>
   <extends>QWidget</extends>
   <header location="global">glwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
